src/message/__tests__/messageActions-test.js

```diff
@@ -0,0 +1,58 @@
+import mockStore from 'redux-mock-store';
+
+import { doNarrow } from '../messagesActions';
+import { streamNarrow, homeNarrow } from '../../utils/narrow';
+
+const narrow = streamNarrow('some stream');
+const streamNarrowStr = JSON.stringify(narrow);
+const homeNarrowStr = JSON.stringify(homeNarrow());
+
+describe('messageActions', () => {
+  describe('doNarrow', () => {
+    test('when no messages in new narrow, actions to fetch messages and switch narrow are dispatched', () => {
```

Comment:
This should depend on caughtUp structure.
If no messages and caughtUp is `{ older: true, newer: true }` no messages are dispatched.